testutils/infra/smtpd_mock.py

```diff
@@ -11,45 +11,48 @@
 #    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
-
-import asyncore
 import base64
 import email
 import imaplib
 import os
 import pytest
 import time
-import smtpd
 import logging
 
-from threading import Thread, Condition
+from aiosmtpd.controller import Controller
+
+from threading import Condition
 
 from redo import retriable
 
 
 class Message:
-    def __init__(self, peer, mailfrom, rcpttos, subject, data):
+    def __init__(self, peer, mail_from, rcpttos, subject, data):
         self.peer = peer
-        self.mailfrom = mailfrom
+        self.mail_from = mail_from
         self.rcpttos = rcpttos
         self.subject = subject
         self.data = data
 
     def __repr__(self):
-        return f"<Message peer={self.peer} mailfrom={self.mailfrom} rcpttos={self.rcpttos} subject={self.subject}>"
+        return f"<Message peer={self.peer} mailfrom={self.mail_from} rcpttos={self.rcpttos} subject={self.subject}>"
```

Comment:
Nice with the property rename. Please propagate it to the representation too:
```suggestion
        return f"<Message peer={self.peer} mail_from={self.mail_from} rcpttos={self.rcpttos} subject={self.subject}>"
```

Reply:
Now that I see the rest of the code, I suggest we also here rename `rcpttos` to `rcpt_tos`